engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/formatter/ConfluentAvroMessageReader.scala

```diff
@@ -1,94 +1,53 @@
-                                                    useStringForKey: Boolean,
-                                                    keySeparator: String)
+private[confluent] class ConfluentAvroMessageReader(schemaRegistryClient: SchemaRegistryClient)
   extends AbstractKafkaAvroSerializer with DatumReaderWriterMixin {
 
   schemaRegistry = schemaRegistryClient
 
-  private val keySubject = ConfluentUtils.keySubject(topic)
-
-  private val valueSubject = ConfluentUtils.valueSubject(topic)
-
   private val decoderFactory = DecoderFactory.get
 
-  // TODO: This implementation won't handle separator escaping
-  def readMessage(str: String, keySchema: Option[Schema], valueSchema: Option[Schema]): ConsumerRecord[Array[Byte], Array[Byte]] = {
+  def readJson[T: ClassTag](jsonObj: Json, schema: Schema, subject: String): Array[Byte] = {
     try {
-      val keyIndex = str.indexOf(keySeparator)
-      if (keyIndex < 0) {
-        throw new SerializationException("No key found in line " + str)
-      } else {
-        val keyString = str.substring(0, keyIndex)
-        val valueString = if (keyIndex + 1 > str.length) "" else str.substring(keyIndex + 1)
-        keySchema match {
-          case None =>
-            val value = jsonToAvro(valueString, valueSchema.get)
-            val serializedValue = serializeImpl(valueSubject, value, new AvroSchema(valueSchema.get))
-            new ConsumerRecord(topic, 0, 0L, Array[Byte](), serializedValue)
-          case Some(_) =>
-            // default scenario when key and value is provided
-            if (keyString.length == 0) {
-              // key schema is provided but there is no content
-              throw new SerializationException("No key content found in line " + str)
-            }
-            val serializedKey = if (useStringForKey) {
-              keyString.getBytes(StandardCharsets.UTF_8)
-            } else {
-              val key = jsonToAvro(keyString, keySchema.get)
-              serializeImpl(keySubject, key, new AvroSchema(keySchema.get))
-            }
-            val value = jsonToAvro(valueString, valueSchema.get)
-            val serializedValue = serializeImpl(valueSubject, value, new AvroSchema(valueSchema.get))
-            new ConsumerRecord(topic, 0, 0L, serializedKey, serializedValue)
-        }
-      }
+      val avroObj = jsonToAvro[T](jsonObj, schema)
+      val serializedValue = serializeImpl(subject, avroObj, new AvroSchema(schema))
+      serializedValue
     } catch {
-      case ex: IOException =>
+      case ex: Exception =>
         throw new SerializationException("Error reading from input", ex)
     }
   }
 
-  private def jsonToAvro(jsonString: String, schema: Schema) = {
+  private def jsonToAvro[T: ClassTag](jsonObj: Json, schema: Schema): AnyRef = {
     try {
-      val reader = createDatumReader(schema, schema, useSchemaReflection = false, useSpecificAvroReader = false)
+      val jsonString = jsonObj.noSpaces
```

Comment:
can you move this line outside the try-catch block and use it in handling exception - will be visible, that this method can't throw exception